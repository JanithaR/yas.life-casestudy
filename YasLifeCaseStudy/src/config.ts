import {Currency} from './interfaces/index';

export const fixerKey: string = '535e747f27804f52b1fe203fda66f299';
export const baseApiUrl: string = 'http://data.fixer.io/api/';

export const currencies: Currency[] = [
	{label: 'Euro', code: 'EUR', pickerValue: 'euro'},
	{label: 'Danmark', code: 'DKK', pickerValue: 'denmark'},
	{label: 'India', code: 'INR', pickerValue: 'india'},
	{label: 'US', code: 'USD', pickerValue: 'us'},
	{label: 'Thailand', code: 'THB', pickerValue: 'thailand'}
];
