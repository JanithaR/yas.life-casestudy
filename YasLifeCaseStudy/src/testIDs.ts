export const FROM_TEXT = 'fromText';
export const TO_TEXT = 'toText';
export const VALUE_INPUT = 'valueInput';
export const CONVERT_BUTTON = 'convertButton';
