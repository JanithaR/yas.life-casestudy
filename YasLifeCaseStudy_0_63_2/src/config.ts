import { Currency } from 'src/interfaces';

export const baseApiUrl: string = 'http://data.fixer.io/api/';
export const latestEndpoint: string = 'latest';
export const currencies: Currency[] = [
    { label: 'Europe', code: 'EUR' },
    { label: 'Sri Lanka', code: 'LKR' },
    { label: 'US', code: 'USD' },
    { label: 'United Arab Emirates', code: 'AED' },
];
