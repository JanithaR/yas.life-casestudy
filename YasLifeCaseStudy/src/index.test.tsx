import React from 'react';
import { currencies } from './config';
import { Provider } from 'react-redux';
import { render, RenderAPI } from 'react-native-testing-library';
import CurrencyConverter from '.';
import { FROM_TEXT, TO_TEXT, VALUE_INPUT } from './testIDs';
import { createStore } from 'redux';
import combinedReducers from './reducers/index';

describe('Currency converter redux connected component', () => {
	let component: RenderAPI;

	beforeEach(() => {
		const mockState = {
			converter: {
				userInput: '1',
				desiredCurrency: currencies[1].code,
				isFetchingLatestRates: false,
				latestRatesFetchError: null,
				latestRatesFetchResponse: null,
				outputs: null
			}
		};

		const mockStore = createStore(combinedReducers, mockState);

		component = render(
			<Provider store={mockStore}>
				<CurrencyConverter />
			</Provider>
		);
	});

	it('should render with the default state from Redux store', () => {
		const { getByText, getByTestId } = component;

		let element = getByText('Click convert');
		expect(element).toBeDefined();

		element = getByTestId(VALUE_INPUT);
		expect(element).toBeDefined();

		expect(component.toJSON()).toMatchSnapshot();

		element = getByTestId(FROM_TEXT);
		expect(element.props.children).toEqual('â‚¬1.00 = ');

		element = getByTestId(TO_TEXT);
		expect(element.props.children).toEqual('DKK\xa00.00');
	});
});
