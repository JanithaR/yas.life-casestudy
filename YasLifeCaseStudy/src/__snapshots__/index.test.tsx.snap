// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency converter redux connected component should render with the default state from Redux store 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "rgb(232, 228, 253)",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "paddingBottom": 10,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 10,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 10,
          "marginBottom": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCompleteType="off"
        autoCorrect={false}
        keyboardType="decimal-pad"
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "rgb(232, 228, 253)",
            "borderRadius": 10,
            "borderWidth": 1,
            "fontSize": 24,
            "height": 40,
            "marginLeft": 5,
            "marginRight": 5,
            "marginTop": 5,
            "padding": 5,
          }
        }
        testID="valueInput"
        underlineColorAndroid="transparent"
        value="1"
      />
      <View>
        <RCTPicker
          items={
            Array [
              Object {
                "label": "Danmark",
                "textColor": undefined,
                "value": "DKK",
              },
              Object {
                "label": "India",
                "textColor": undefined,
                "value": "INR",
              },
              Object {
                "label": "US",
                "textColor": undefined,
                "value": "USD",
              },
              Object {
                "label": "Thailand",
                "textColor": undefined,
                "value": "THB",
              },
              Object {
                "label": "Sri Lanka",
                "textColor": undefined,
                "value": "LKR",
              },
            ]
          }
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              undefined,
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "height": 50,
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="convertButton"
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "rgb(95, 80, 181)",
                  },
                ]
              }
            >
              Convert
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "padding": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(95, 80, 181, 0.8)",
            "fontSize": 24,
          }
        }
        testID="fromText"
      >
        €1.00 = 
      </Text>
      <Text
        style={
          Object {
            "color": "rgb(95, 80, 181)",
            "fontSize": 36,
          }
        }
        testID="toText"
      >
        DKK 0.00
      </Text>
      <Text
        style={
          Object {
            "fontSize": 12,
          }
        }
      >
        Click convert
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
